lib/components.js

```diff
@@ -363,7 +363,9 @@ class DatePicker extends Component {
       'minimumDate',
       'minuteInterval',
       'mode',
-      'timeZoneOffsetInMinutes'
+      'timeZoneOffsetInMinutes',
+      'background',
```

Comment:
I can't tell right now which one I would pick, both have advantages and downsides. I'm always in favor of removing nesting, so I like how it's in this PR. However, having two separate ``ios`` and ``android`` objects to configure their respective props would make more easy to set different values for same properties under the two platforms. This is something I've been doing with ``Platform.OS`` for `keyboardType`, for instance.

I would pick either having all the things inside ``ios`` or `android` or not having those  objects. It's easier to reason about and less error prone than having the common options and then some custom options per platform inside the objects IMHO. Whatever solution we choose, should be implemented in all components, no only in this one.

What I would change before merging this code is to remove the `config` object in favor of using the `mode` option of `DatePickerIOS` to set the time or date picker in Android (both do conceptually the same), and also put the `format` option in the same level. We could have some name collision risk, but is highly unlikely (a new native module version should have that property first and then it should be implemented inside FB RN component) and we always can add a backslash (`_format`) or something like a _namespace_ (`tcombFormat`) to avoid that. In fact, that could be a good way to define custom functions for the components in the future.